tests/integration/ha_tests/test_smoke.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from asyncio import TimeoutError
+
+import pytest
+from juju import tag
+from pytest_operator.plugin import OpsTest
+from tenacity import Retrying, stop_after_delay, wait_fixed
+
+from ..helpers import (
+    APPLICATION_NAME,
+    CHARM_SERIES,
+    get_primary,
+)
+from .helpers import (
+    check_db,
+    check_password_auth,
+    create_db,
+    get_any_deatached_storage,
+    is_postgresql_ready,
+    is_storage_exists,
+    storage_id,
+)
+
+TEST_GARBADGE_STORAGE_NAME = "test_pgdata"
+TEST_DATABASE_RELATION_NAME = "test_database"
+DUP_APPLICATION_NAME = "postgres-test-dup"
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_removal(ops_test: OpsTest, charm: str):
+    """Test all recoureces is removed after application removal."""
+    # Deploy the charm.
+    async with ops_test.fast_forward():
+        await ops_test.model.deploy(
+            charm,
+            application_name=APPLICATION_NAME,
+            num_units=1,
+            series=CHARM_SERIES,
+            config={"profile": "testing"},
+        )
+
+        # Reducing the update status frequency to speed up the triggering of deferred events.
+        await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1500)
+        assert ops_test.model.applications[APPLICATION_NAME].units[0].workload_status == "active"
+
+        primary_name = await get_primary(
+            ops_test, ops_test.model.applications[APPLICATION_NAME].units[0].name
+        )
+        assert await is_postgresql_ready(ops_test, primary_name)
+
+        storage_id_str = storage_id(ops_test, primary_name)
+
+        # Check if storage exists after application deployed
+        assert await is_storage_exists(ops_test, storage_id_str)
+
+        await ops_test.model.remove_application(
+            APPLICATION_NAME, block_until_done=True, destroy_storage=True
+        )
+
+        # Check if storage removed after application removal
+        assert not await is_storage_exists(ops_test, storage_id_str)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_force_removal(ops_test: OpsTest, charm: str):
```

Comment:
IMHO there is no need to test this either. With force removal the charm code shouldn't even fire, so this would be handled by Juju entirely.

Reply:
@taurus-forever suggested to leave `test_app_force_removal` just for the sake of testing the correctness of storage detaching (charm will be removed with force correctly). Since this is smoke test it should be fast.